packages/components/src/custom-select-control-v2/stories/index.story.tsx

```diff
@@ -0,0 +1,149 @@
+/**
+ * External dependencies
+ */
+import type { Meta, StoryFn } from '@storybook/react';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { CustomSelect, CustomSelectItem } from '..';
+
+const meta: Meta< typeof CustomSelect > = {
+	title: 'Components (Experimental)/CustomSelectControl v2',
+	component: CustomSelect,
+	subcomponents: {
+		// @ts-expect-error - See https://github.com/storybookjs/storybook/issues/23170
+		CustomSelectItem,
+	},
+	argTypes: {
+		children: { control: { type: null } },
+	},
+	parameters: {
+		actions: { argTypesRegex: '^on.*' },
+		controls: { expanded: true },
+		docs: {
+			canvas: { sourceState: 'shown' },
+			source: { excludeDecorators: true },
+		},
+	},
+	decorators: [
+		( Story ) => (
+			<div
+				style={ {
+					minHeight: '250px',
+				} }
+			>
+				<Story />
+			</div>
+		),
+	],
+};
+export default meta;
+
+const Template: StoryFn< typeof CustomSelect > = () => {
+	return (
+		<CustomSelect label="Label" defaultValue="Default">
+			<CustomSelectItem value="Small">
+				<span style={ { fontSize: '75%' } }>Small</span>
+			</CustomSelectItem>
+			<CustomSelectItem value="Default">Default</CustomSelectItem>
+			<CustomSelectItem value="Large">
+				<span style={ { fontSize: '150%' } }>Large</span>
+			</CustomSelectItem>
+			<CustomSelectItem value="Huge">
+				<span style={ { fontSize: '200%' } }>Huge</span>
+			</CustomSelectItem>
+		</CustomSelect>
+	);
+};
+
+export const Default = Template.bind( {} );
+
+const MultiSelectTemplate: StoryFn< typeof CustomSelect > = () => {
+	function renderValue( value: string | string[] ) {
+		if ( value.length === 0 ) return '0 colors selected';
+		return <div>{ value.length } colors selected</div>;
+	}
+
+	const items = [
+		'amber',
+		'aquamarine',
+		'flamingo pink',
+		'lavendar',
+		'maroon',
+		'tangerine',
+	];
+
+	const [ value, setValue ] = useState< string | string[] >( [
+		'lavendar',
+		'tangerine',
+	] );
+
+	return (
+		<CustomSelect
+			onChange={ ( nextValue ) => setValue( nextValue ) }
+			defaultValue={ value }
+			label="Select Colors"
+			renderSelectedValue={ ( currentValue ) =>
+				renderValue( currentValue )
+			}
+		>
+			{ items.map( ( item ) => (
+				<CustomSelectItem key={ item } value={ item }>
+					{ item }
+				</CustomSelectItem>
+			) ) }
+		</CustomSelect>
+	);
+};
+
+export const MultiSelect = MultiSelectTemplate.bind( {} );
+
+const ControlledTemplate = () => {
+	function renderValue( gravatar: string | string[] ) {
+		const avatar = `https://gravatar.com/avatar?d=${ gravatar }`;
+		return (
+			<div style={ { display: 'flex', alignItems: 'center' } }>
```

Comment:
This div with flex styles was added after adding the checkmark. The `CustomSelectItem` needed `display: flex` so the checkmark is rendered correctly (aligning with our current `CustomSelectControl`): 

https://github.com/WordPress/gutenberg/blob/3b5f968baf20425518ba6232aa87303342ab1eb8/packages/components/src/custom-select-control-v2/styles.ts#L56-L59

It doesn't seem ideal, but the checkmark is rendered as another child. So I can see potential overrides happening unless the consumer wraps the content in a div, like I did above.